src/Catalyst.Node.POA.CE.Tests/IntegrationTests/PoaTestNode.cs

```diff
@@ -0,0 +1,177 @@
+#region LICENSE
+
+/**
+* Copyright (c) 2019 Catalyst Network
+*
+* This file is part of Catalyst.Node <https://github.com/catalyst-network/Catalyst.Node>
+*
+* Catalyst.Node is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 2 of the License, or
+* (at your option) any later version.
+*
+* Catalyst.Node is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Catalyst.Node. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Autofac;
+using Catalyst.Abstractions;
+using Catalyst.Abstractions.Consensus;
+using Catalyst.Abstractions.Cryptography;
+using Catalyst.Abstractions.Dfs;
+using Catalyst.Abstractions.FileSystem;
+using Catalyst.Abstractions.KeySigner;
+using Catalyst.Abstractions.Keystore;
+using Catalyst.Abstractions.Mempool;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Abstractions.P2P.Discovery;
+using Catalyst.Abstractions.Rpc;
+using Catalyst.Abstractions.Types;
+using Catalyst.Core.Lib.Config;
+using Catalyst.Core.Lib.DAO;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Lib.FileSystem;
+using Catalyst.Core.Lib.P2P.Models;
+using Catalyst.Core.Lib.P2P.Repository;
+using Catalyst.Core.Modules.Dfs;
+using Catalyst.Core.Modules.Hashing;
+using Catalyst.Core.Modules.Mempool;
+using Catalyst.Core.Modules.Rpc.Server;
+using Catalyst.Core.Modules.Web3;
+using Catalyst.Protocol.Network;
+using Catalyst.Protocol.Peer;
+using Catalyst.TestUtils;
+using NSubstitute;
+using SharpRepository.InMemoryRepository;
+using TheDotNetLeague.MultiFormats.MultiHash;
+using Xunit.Abstractions;
+
+namespace Catalyst.Node.POA.CE.Tests.IntegrationTests
+{
+    public class PoaTestNode : ICatalystNode, IDisposable
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fix this implementation of 'IDisposable' to conform to the dispose pattern.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4594394)